source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py

```diff
@@ -275,29 +275,47 @@ def build_ridesharing_journeys(self, from_pt_obj, to_pt_obj, period_extremity):
             dropoff_coord = get_pt_object_coord(pb_rsj_dropoff)
 
             pb_rsj.requested_date_time = period_extremity.datetime
-            if rsj.pickup_date_time:
-                pb_rsj.departure_date_time = rsj.pickup_date_time
-            if rsj.dropoff_date_time:
-                pb_rsj.arrival_date_time = rsj.dropoff_date_time
+            if rsj.departure_date_time:
+                pb_rsj.departure_date_time = rsj.departure_date_time
+            if rsj.arrival_date_time:
+                pb_rsj.arrival_date_time = rsj.arrival_date_time
             pb_rsj.tags.append('ridesharing')
 
             # start teleport section
             start_teleport_section = pb_rsj.sections.add()
             start_teleport_section.id = "section_{}".format(six.text_type(generate_id()))
-            start_teleport_section.type = response_pb2.CROW_FLY
             start_teleport_section.street_network.mode = response_pb2.Walking
             start_teleport_section.origin.CopyFrom(from_pt_obj)
             start_teleport_section.destination.CopyFrom(pb_rsj_pickup)
-            start_teleport_section.length = int(crowfly_distance_between(from_coord, pickup_coord))
+            if rsj.origin_pickup_shape:
+                if len(rsj.origin_pickup_shape) == 2:
```

Comment:
Add a comment "If stree_network shape in the offer contains with only two points use default CROW_FLY section"